### PR TITLE
A small refactor of the e2e tests

### LINKED ISSUE #3724: Refactor e2e tests to make functions unexported
Most functions in the e2e tests ended up being exported when they don't need to be.

**Vote on this issue!**

This is an invitation to the Velero community to vote on issues, you can see the project's [top voted issues listed here](https://github.com/vmware-tanzu/velero/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc).  
Use the "reaction smiley face" up to the right of this comment to vote.

- :+1: for "I would like to see this bug fixed as soon as possible"
- :-1: for "There are more important bugs to focus on right now"

### PR DESCRIPTION
**Pull request by carlisia**
Signed-off-by: Carlisia <carlisia@grokkingtech.io>

A small refactor to make functions private since they are not at the moment needed to be exported. We can export functions as we start moving things into separate packages.

Fixes #3724.

# Please indicate you've done the following:

- [x] [Accepted the DCO](https://velero.io/docs/v1.5/code-standards/#dco-sign-off). Commits without the DCO will delay acceptance.
- [x] [Created a changelog file](https://velero.io/docs/v1.5/code-standards/#adding-a-changelog) or added `/kind changelog-not-required`.
- [ ] Updated the corresponding documentation in `site/content/docs/main`.

### CODE REVIEW

**Review by dsu-igeek — APPROVED** [review]
LGTM

**Review by zubron — DISMISSED** [review]
Thanks!

**Comment by zubron** [comment]
Hmm, the [CI check is failing](https://github.com/vmware-tanzu/velero/pull/3726/checks?check_run_id=2415925592) due to differences in the go.mod and go.sum. Can you run `make ci` and update this? Thanks!

**Comment by carlisia** [comment]
ooops! I'm terribly sorry. All fixed.

File: pkg/client/factory.go [hunk 1]
```diff
@@ -151,6 +154,8 @@ func (f *factory) KubebuilderClient() (kbclient.Client, error) {
 
 	scheme := runtime.NewScheme()
 	velerov1api.AddToScheme(scheme)
+	k8scheme.AddToScheme(scheme)
```
Comment (zubron) [inline pkg/client/factory.go:157]: Were these changes necessary as part of the CI fix?

Comment (carlisia) [reply pkg/client/factory.go]: Good catch @zubron. No, this was supposed to be part of another branch for an incoming change, but not this one. Thank you, reverted.

**Comment by carlisia** [comment]
Had to revert a change, ready for review again. Thanks.

File: test/e2e/client.go [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+import (
+	"k8s.io/client-go/kubernetes"
+	kbclient "sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/vmware-tanzu/velero/pkg/client"
+)
+
+// testClient contains different API clients that are in use throughout
+// the e2e tests.
+type testClient struct {
```
Comment (zubron) [inline test/e2e/client.go:28]: Ah, sorry I didn't include this in the last review! Is this also part of the follow up refactor branch? I can't see it being used here yet.

Comment (carlisia) [reply test/e2e/client.go:28]: Yep. Note to self: don't do things in a hurry. Or maybe don't do anything today. Argh!

**Review by zubron — APPROVED** [review]
LGTM!

**Review by dsu-igeek — APPROVED** [review]
LGTM

**Pull request merged**